src/merge.rs

```diff
@@ -192,3 +194,295 @@ impl<'repo> Drop for AnnotatedCommit<'repo> {
+            .unwrap_or(ptr::null());
+
+        self
+    }
+
+    /// Specify ancestor label, default is "theirs"
+    pub fn their_label<T: IntoCString>(&mut self, t: T) -> &mut MergeFileOptions {
+        self.their_label = Some(t.into_c_string().unwrap());
+
+        self.raw.their_label = self
+            .their_label
+            .as_ref()
+            .map(|s| s.as_ptr())
+            .unwrap_or(ptr::null());
+
+        self
+    }
+
+    /// Specify a side to favor for resolving conflicts
+    pub fn file_favor(&mut self, favor: FileFavor) -> &mut MergeFileOptions {
+        self.raw.favor = favor.convert();
+        self
+    }
+
+    /// Specify marker size, default is 7: <<<<<<< ours
+    pub fn marker_size(&mut self, size: u16) -> &mut MergeFileOptions {
+        self.raw.marker_size = size;
+        self
+    }
+
+    /// Acquire a pointer to the underlying raw options.
+    pub unsafe fn raw(&self) -> *const raw::git_merge_file_options {
+        &self.raw as *const _
+    }
+}
+
+/// For git_merge_file_input
+pub struct MergeFileInput {
+    raw: raw::git_merge_file_input,
+
+    /// File name of the conflicted file, or `NULL` to not merge the path.
+    ///
+    /// You can turn this value into a `std::ffi::CString` with
+    /// `CString::new(&entry.path[..]).unwrap()`. To turn a reference into a
+    /// `&std::path::Path`, see the `bytes2path()` function in the private,
+    /// internal `util` module in this crate’s source code.
+    path: Option<CString>,
+
+    /// File mode of the conflicted file, or `0` to not merge the mode.
+    pub mode: Option<FileMode>,
+
+    /// File content
+    pub content: Option<Vec<u8>>,
+}
+
+impl Default for MergeFileInput {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl MergeFileInput {
+    /// Creates a new set of empty diff options.
+    pub fn new() -> MergeFileInput {
+        let mut input = MergeFileInput {
+            raw: unsafe { mem::zeroed() },
+            path: None,
+            mode: None,
+            content: None,
+        };
+        assert_eq!(
+            unsafe { raw::git_merge_file_input_init(&mut input.raw, 1) },
+            0
+        );
+        input
+    }
+
+    /// File name of the conflicted file, or `None` to not merge the path.
+    pub fn path<T: IntoCString>(&mut self, t: T) -> &mut MergeFileInput {
+        self.path = Some(t.into_c_string().unwrap());
+
+        self.raw.path = self
+            .path
+            .as_ref()
+            .map(|s| s.as_ptr())
+            .unwrap_or(ptr::null());
+
+        self
+    }
+
+    /// File mode of the conflicted file, or `0` to not merge the mode.
+    pub fn mode(&mut self, mode: Option<FileMode>) -> &mut MergeFileInput {
+        self.mode = mode;
+
+        if let Some(mode) = self.mode {
+            self.raw.mode = mode as u32;
+        }
+
+        self
+    }
+
+    /// File content, text or binary
+    pub fn content(&mut self, content: Option<Vec<u8>>) -> &mut MergeFileInput {
```

Comment:
I think it might actually be best to take `&[u8]` here and tie the lifetime to `MergeFileInput` so that way users won't have to copy data around.

Reply:
I have made it using `lifetime`, but the `merge_file()` in `repo.rs`(line 1868) becomes ugly, could you tell me how to improve it?